controllers/topology/reconcile_state.go

```diff
@@ -62,17 +66,158 @@ func (r *ClusterReconciler) reconcileCluster(ctx context.Context, current, desir
 	// Check differences between current and desired state, and eventually patch the current object.
 	patchHelper, err := newMergePatchHelper(current.cluster, desired.cluster, r.Client)
 	if err != nil {
-		return errors.Wrap(err, "failed to create patch helper for the Cluster object")
+		return errors.Wrapf(err, "failed to create patch helper for %s %s", current.cluster.GroupVersionKind(), current.cluster.Name)
```

Comment:
```suggestion
		return errors.Wrapf(err, "failed to create patch helper for %s/%s", current.cluster.GroupVersionKind(), current.cluster.Name)
```
as per https://github.com/kubernetes-sigs/cluster-api/issues/5067#issuecomment-896857120 (same for the other error messages)

Reply:
Ah, I didn't take the `gvk/name` format literally :) But sounds like a good idea. Will update accordingly